src/controllers/cart.controller.ts

```diff
@@ -0,0 +1,218 @@
+// src/cart/cart.controller.ts
+
+import type { Prisma } from '@prisma/client';
+import type { NextFunction, Request, Response } from 'express';
+
+import type { PatchCartInput } from '../dtos/cart.dto.js';
+import { getCartItemWithDetails, getCartWithItems } from '../repositories/cart.repository.js';
+import * as cartService from '../services/cart.service.js';
+
+type SizeDetail = { ko?: string; en?: string; [key: string]: unknown };
+
+const toSizeDetail = (value: Prisma.JsonValue | null): SizeDetail | null => {
+  if (!value || typeof value !== 'object' || Array.isArray(value)) return null;
+  return value as SizeDetail;
+};
+
+// 공통 401 응답 헬퍼 (선택이지만 편해서 추가)
+function getUserIdOr401(req: Request, res: Response): string | null {
+  const user = req.user;
+  if (!user) {
+    res.status(401).json({
+      statusCode: 401,
+      message: '인증이 필요합니다.',
+      error: 'Unauthorized',
```

Comment:
```suggestion
    throw HttpException.userNotFound()
```
에러 공통 처리를 위해서 HttpException과 미들웨어가 만들어져 있는 상태라서 HttpException 호출해서 사용하면 좋을 것 같습니다 !